examples/kafkamgmt/kafka_mgmt.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	kafkamgmt "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1"
+
+	"golang.org/x/oauth2"
+)
+
+func main() {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
```

Comment:
@wtrocki I spent a lot of time this afternoon to try and see if we could use the Golang OAuth packages to auto-refresh the tokens. This is supposed to be possible with https://pkg.go.dev/golang.org/x/oauth2#Config.TokenSource but I was unsuccessful unfortunately :(

It is one reason why I added https://github.com/redhat-developer/app-services-sdk-go/blob/wrappers/endpoint.go.

Do you have any experience with using this? Here's what I did:

```go
ctx := context.Background()
	tk := oauth2.Token{
		AccessToken:  os.Getenv("ACCESS_TOKEN"),
		RefreshToken: os.Getenv("REFRESH_TOKEN"),
	}
	oauthCfg := oauth2.Config{
		Endpoint: rhoas.Endpoint,
		ClientID: "rhoas-cli-prod",
		Scopes:   []string{"openid"},
	}

	tc := oauth2.NewClient(ctx, oauthCfg.TokenSource(ctx, &tk))

	client := kafkamgmt.NewAPIClient(&kafkamgmt.Config{
		HTTPClient: tc,
	})
```

I thought by doing this the token would refresh automatically if the access token is expired (important to have for interactive mode where the user could have left their terminal idle). But I get 401. Any ideas what I could be doing wrong here?

Reply:
I thought that we keep auth off the SDKs for now

Reply:
> I thought that we keep auth off the SDKs for now.

Yeah, well this is just showing how to configure the auth header for a request - it is not adding an auth SDK package or anything.

Reply:
I'm not sure how this works with golang SDK. In JS-SDK I can pass token string and all implementors simply initialize SDK on request (rather than have it as singleton etc.)